weblab/experiments/views.py

```diff
@@ -52,19 +49,21 @@ class ExperimentTasks(LoginRequiredMixin, ListView):
     Show running and queued versions of all experiments
     Delete checked versions
     """
-    model = Experiment
+    model = RunningExperiment
     template_name = "experiments/experiment_tasks.html"
 
     def get_queryset(self):
-        return self.model.objects.filter(
-            Q(author=self.request.user) &
-            Q(versions__status=ExperimentVersion.STATUS_QUEUED) |
-            Q(versions__status=ExperimentVersion.STATUS_RUNNING))
+        q_running_experiments = RunningExperiment.objects.filter(
+            Q(experiment_version__author=self.request.user)
+        ).order_by(
+            'experiment_version__created_at',
+        ).select_related('experiment_version')
+        return q_running_experiments
 
     def post(self, request):
         """Process list of user selected checkboxes and kill running versions"""
         for ver in request.POST.getlist('chkBoxes[]'):
-            self.model.objects.filter(versions=ver).delete()
+            self.model.objects.filter(experiment_version_id=ver).delete()
```

Comment:
I think this now wouldn't work, as self.model is `RunningExperiment` but the ids you're getting back are the `ExperimentVersion` ids. So probably need `ExperimentVersion.objects.filter`?

Reply:
thanks, It did seem to work ok.... Perhaps because the experiment version id field is also on the running experiemnts table? however your way is cleaner and more readable so I have implemented your suggestion here

Reply:
Indeed, because all id fields are just sequences of integers, I suspect in the test both happened to be 1!